### PR TITLE
feat: implement name property for the member object

### PR DESCRIPTION
**Pull request by sillyElvy**
## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [ ] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent): #746 
- [ ] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/api/models/member.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def __init__(self, **kwargs):
             self.avatar = self.user.avatar
 
     def __repr__(self) -> str:
-        return self.user.username if self.user else self.nick
+        return self.nick or (self.user.username if self.user else None)
```
Comment (Toricane) [inline interactions/api/models/member.py]: Can't you just do `return self.name` here?

File: interactions/api/models/member.py [hunk 2]
```diff
@@ -105,6 +105,16 @@ def mention(self) -> str:
         """
         return f"<@!{self.user.id}>" if self.nick else f"<@{self.user.id}>"
 
+    @property
+    def name(self) -> str:
+        """
+        Returns a string of whether the users nickname or its username, if a nickname isn't present
```
Comment (Toricane) [inline interactions/api/models/member.py]: I think this should be:
`Returns the string of either the user's nickname or username.`
The one you had was confusing and with some grammar mistakes.
BTW you were also missing a period at the end.

**Review by Toricane — CHANGES_REQUESTED** [review]
Looks good, just a couple of things.

File: interactions/api/models/member.py (hunk 1 above)
Comment (sillyElvy) [reply interactions/api/models/member.py]: Does that exist?

File: interactions/api/models/member.py (hunk 2 above)
Comment (sillyElvy) [reply interactions/api/models/member.py]: I see the grammar mistake now lol

I had like 5 minutes of my computer time left and wanted to finish it :lol:

Comment (sillyElvy) [reply interactions/api/models/member.py]:
```suggestion
        returns the string of either the user's nickname or username
```

testing if this works, I deleted the fork..

File: interactions/api/models/member.py [hunk 3]
```diff
@@ -105,6 +105,16 @@ def mention(self) -> str:
         """
         return f"<@!{self.user.id}>" if self.nick else f"<@{self.user.id}>"
 
+    @property
+    def name(self) -> str:
+        """
+        returns the string of either the user's nickname or username
```
Comment (sillyElvy) [inline interactions/api/models/member.py]:
```suggestion
        Returns the string of either the user's nickname or username.
```

File: interactions/api/models/member.py (hunk 1 above)
Comment (Toricane) [reply interactions/api/models/member.py]: > Does that exist?

You yourself implemented it in this PR.

Comment (sillyElvy) [reply interactions/api/models/member.py]: OH GOD I'M STUPID

Comment (sillyElvy) [inline interactions/api/models/member.py]:
```suggestion
        return self.name
```

Comment (Toricane) [reply interactions/api/models/member.py]: To clarify even more, this line is the same as your `name` property that you implemented, so why not just do:
```py
def __repr__(self) -> str:
    return self.name
```

Comment (sillyElvy) [reply interactions/api/models/member.py]: Yeah I got it

**Review by Toricane — APPROVED** [review]
LGTM!

**Pull request merged**